wordpress-mobile/WordPress-Android
# Quick Start for Existing Users V2: Add new quick start task type

Parent #16350

Corresponding FluxC PR: https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2376

This PR completes when branches for the newly added quick start task type / dynamic card type in `FluxC` so that the code compiles.

To test:
- Launch the app with a site having a quick start in progress.
- Make sure that tasks for only `Customize` and `Grow` quick start types are shown in the "Next Steps" card on `My Site` screen. Note that "Next Steps" card will be present either on `Home` or `Menu` tab based on `Initial Screen` settings.
- Go to `My Site` -> `Me` -> `App Settings` -> `Debug` settings and enable `QuickStartDynamicCardsFeatureConfig`.
- Restart the app.
- Notice that dynamic cards are displayed only for dynamic card types corresponding to `Customize` and `Grow` tasks.

P.S.: I've left a few todos to add tracking and category completion message for the newly added task type. Added a task to the parent task to remove these todos.

Merge Instructions

1. Make sure corresponding `FluxC` PR is merged to trunk (I can take care of merging and updating version) ✅ 
2. Update `FluxC` version ✅ 
3. Wait for known issue in `Regression Notes` to be fixed ✅ - 856e756
4. Remove `Not Ready for Merge` label ✅ 
5. Merge the PR ✅ 

## Regression Notes
1. Potential unintended areas of impact
[A] Crashes or empty qs card (without tasks) display due to the introduction of a new task/ dynamic card type.
[B]  Impact on Quick Start completion. 

2. What I did to test those areas of impact (or what existing automated tests I relied on)
- For [A] - tested manually (See `To test` section) 
- For [B] - verified that [isEveryQuickStartTaskDone()](https://github.com/wordpress-mobile/WordPress-Android/blob/082b8f039aef4abf647e9b927bd3d3834137f25e/WordPress/src/main/java/org/wordpress/android/util/QuickStartUtils.kt#L205-L208) checks for done status for `QuickStartTask.values()`  which do not contain any tasks for the newly added task type yet.

    Known Issue: Completion notification is shown for dynamic cards on selecting a site from Login Epilogue.

3. What automated tests I added (or what prevented me from doing so)
- Fixed `FluxC` existing tests for `DynamicCardStore`.

PR submission checklist:
